src/unify/mod.rs

```diff
@@ -299,6 +299,16 @@ impl<S: UnificationStore> UnificationTable<S> {
         self.values.len()
     }
 
+    /// Returns the keys of all variables created since the `snapshot`.
+    pub fn vars_since_snapshot(
+        &self,
+        snapshot: &Snapshot<S>,
+    ) -> impl Iterator<Item = S::Key> + '_ {
+        self.values.values_since_snapshot(&snapshot.snapshot).map(move |idx| {
+            self.values[idx].parent
```

Comment:
Wait, this is wrong, isn't it? `parent` is (I think) the unification parent, which would mean this method may return variables older than the snapshot, and may miss some of the newer variables (which, while unified in the unification table, may still be found in values that haven't been normalized since the unification).

You should probably return `Range<S::Key>` by using `UnifyKey::from_index`, AFAICT (look for other uses of `from_index` (I assume some `u32` check may be involved?)

Reply:
@nikomatsakis Why doesn't `ena` rely entirely on `usize` and leave `u32` (or smaller) key handling to the users?

Reply:
Convenience for end-users, I guess. But I think the code you highlighted here @eddyb is outdated, no?